[FIX] purchase: merge RFQ if same currency and merge order lines in appropriate RFQ

nim-odoo opened the pull request:
- RFQ of same currency are merged (fixes #5901 and opw: 632594)
- Merge order lines in the appropriate RFQ

beledouxdenis commented:
I have four remarks
 - I do not understand how this whole things worked before your change to switch the `order_lines_to_move` variable from a `list` to a `dict`. Do you ?
 - I believe you can change your code section:
```
if order_key in order_lines_to_move:    
    order_lines_to_move[order_key] += [order_line.id for order_line in porder.order_line]
else:
    order_lines_to_move[order_key] = [order_line.id for order_line in porder.order_line]
```
with a setdefault:
```
order_lines_to_move.setdefault(order_key, []) += [order_line.id for order_line in porder.order_line]
```
Which has the same behavior, but within one line.

- The typos changes should be done in another commit, as this is out of scope of your fix. I won't oblige you to do it for this PR, but consider this for your future PR.

- Removing the line from the `addons/purchase/wizard/purchase_order_group_view.xml` is useless :).

nim-odoo commented:
- Yes, I do understand. Let's take an example:
RFQ 1 (currency EUR), containing line 1 & 2
RFQ 2 (currency EUR), containing line 3 & 4
RFQ 3 (currency USD), containing line 5 & 6
RFQ 4 (currency USD), containing line 7 & 8

These 4 RFQs are merged. Before the fix, we get:
RFQ 5 (currency EUR), no line attached.
RFQ 6 (currency USD), containing lines 1 to 8.

After the fix, we get:
RFQ 5 (currency EUR), containing lines 1 to 4
RFQ 6 (currency USD), containing lines 5 to 8

- Thanks, although 2 lines are necessary ;)
- OK, removed
- Done automatically by the editor, removed

beledouxdenis commented:
I did not made myself clear regarding my first point: why isn't it enough to just add the `'currency_id'` within the tuple of the `make_key` method call ? Why is the `dict` necessary compared to before  with the `list` ?
